crypto/bn/bn_dsa.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <stdio.h>
+/* The line above prevents "ISO C forbids an empty translation unit" warning */
+#ifndef FIPS_MODE
+# include <string.h>
+# include <openssl/evp.h>
+# include <openssl/rand_drbg.h>
+# include "internal/cryptlib.h"
+# include "internal/bn_int.h"
+# include "../rand/rand_lcl.h"
+
+typedef struct drbg_ctx_st
+{
+    const unsigned char* x;
+    const unsigned char* h;
+    size_t rlen;
+} DRBG_CTX;
```

Comment:
> Could this please be renamed dsa_drbg_ctx_st?

which would imply that `DRBG_CTX` should be renamed to `DSA_DRBG_CTX` as well.

Reply:
Renamed